### PR TITLE
[EDMTP-303] Regression and backward compatibility fixes

### PR DESCRIPTION
**Pull request by smiakchilo**

### CODE REVIEW

File: aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/target/TargetImpl.java [hunk 1]
```diff
@@ -169,8 +177,11 @@ private Target getTarget(String path, boolean createIfMissing) {
         if (StringUtils.isBlank(path)) {
             return null;
         }
-        if (path.contains(DialogConstants.PATH_SEPARATOR)) {
-            Queue<String> pathChunks = Pattern.compile("/").splitAsStream(path).collect(Collectors.toCollection(LinkedList::new));
+        String effectivePath = StringUtils.strip(path, DialogConstants.DOUBLE_QUOTE);
+        boolean isEscaped = !StringUtils.equals(path, effectivePath);
+
+        if (!isEscaped && effectivePath.contains(DialogConstants.PATH_SEPARATOR)) {
+            Queue<String> pathChunks = Pattern.compile("/").splitAsStream(effectivePath).collect(Collectors.toCollection(LinkedList::new));
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/target/TargetImpl.java:184]: (cosmetics) Pattern.compile(PATH_SEPARATOR)?

File: aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/target/TargetImpl.java [hunk 2]
```diff
@@ -182,20 +193,61 @@ private Target getTarget(String path, boolean createIfMissing) {
...
+        collectTargets(this, filter, result);
+        return result;
+    }
+
+    private void collectTargets(Target current, Predicate<Target> filter, List<Target> collection) {
+        List<Target> matches = current.getChildren().stream().filter(filter).collect(Collectors.toList());
+        collection.addAll(matches);
+        for (Target match : matches) {
+            collectTargets(match, filter, collection);
+        }
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/target/TargetImpl.java:233]: please review this algorithm

File: aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/util/PluginReflectionUtility.java [hunk 3]
```diff
@@ -370,23 +371,31 @@ private static void populateRuntimeContext(Object handler, Field field) {
...
-        while (finalList.size() != list.size()) {
-            if (sortedList.get(k).equals(_Default.class)) {
-                k++;
+    /**
+     * Retrieves values of public constant fields originating from the given constant class as a key-value map
+     * @param value {@code Class<?>} object representing the constants class
+     * @return {@code Map<String, Object>} containing the key-value pairs. An empty map can be returned if no valid
+     * constants found
+     */
+    public static Map<String, Object> getConstantValues(Class<?> value) {
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/util/PluginReflectionUtility.java:380]: (cosmetic) perhaps 'value' parameter can be renamed to something else ('constnatsClaass' or 'cls', for example), as it makes line 390 а little bit more confusing that it should be :)

File: aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/target/TargetImpl.java (hunk 1 above)
Comment (smiakchilo) [reply aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/target/TargetImpl.java:184]: Exactly, thank you

File: aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/target/TargetImpl.java (hunk 2 above)
Comment (smiakchilo) [reply aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/target/TargetImpl.java:233]: Fixed, thanks for noticing

File: aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/util/PluginReflectionUtility.java (hunk 3 above)
Comment (smiakchilo) [reply aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/util/PluginReflectionUtility.java:380]: You are right

**Review by liubou-masiuk — APPROVED** [review]

**Review by liubou-masiuk — APPROVED** [review]

**Pull request merged**